src/python/pants/backend/jvm/subsystems/dependency_context.py

```diff
@@ -86,6 +86,8 @@ def compute_fingerprint(self, target):
 
     hasher = hashlib.sha1()
     hasher.update(target.payload.fingerprint().encode())
+    # Adding tags into cache key because it may decide which workflow applies to the target.
+    hasher.update(','.join(sorted(target.tags)).encode())
```

Comment:
Could we please use `pants.base.hash_utils.stable_json_sha1()` here instead?

Reply:
`target.tags` is a set, so should be correct?

Reply:
Yes, but using `stable_json_sha1()` every time we want to take a fingerprint of something makes it a lot easier to debug caching errors afterwards, I think.

Reply:
Thanks for clarifying!